paddle/fluid/framework/fleet/fleet_wrapper.h

```diff
@@ -0,0 +1,131 @@
+// Param<in>: scope, table_id, var_names
+// Param<out>: push_sparse_status
+
+class FleetWrapper {
+ public:
+  FleetWrapper() {}
+  virtual ~FleetWrapper() {}
+
+  // Pull sparse variables from server in Sync mode
+  // Param<in>: scope, table_id, var_names, fea_keys
+  // Param<out>: fea_values
+  void PullSparseVarsSync(const Scope& scope, const uint64_t table_id,
+                          const std::vector<std::string>& var_names,
+                          std::vector<uint64_t>* fea_keys,
+                          std::vector<std::vector<float>>* fea_values,
+                          int fea_dim);
+
+  void PullDenseVarsSync(const Scope& scope, const uint64_t table_id,
+                         const std::vector<std::string>& var_names);
+
+  void PullDenseVarsAsync(
+      const Scope& scope, const uint64_t table_id,
+      const std::vector<std::string>& var_names,
+      std::vector<::std::future<int32_t>>* pull_dense_status);
+
+  // Push dense variables to server in async mode
+  // Param<in>: scope, table_id, var_names,
+  // Param<out>: push_sparse_status
+  void PushDenseVarsAsync(
+      const Scope& scope, const uint64_t table_id,
+      const std::vector<std::string>& var_names,
+      std::vector<::std::future<int32_t>>* push_sparse_status);
+
+  // Push sparse variables with labels to server in Async mode
+  // This is specially designed for click/show stats in server
+  // Param<in>: scope, table_id, var_grad_names,
+  //            fea_keys, fea_labels, sparse_grad_names
+  // Param<out>: push_values, push_sparse_status
+  void PushSparseVarsWithLabelAsync(
+      const Scope& scope, const uint64_t table_id,
+      const std::vector<uint64_t>& fea_keys,
+      const std::vector<float>& fea_labels,
+      const std::vector<std::string>& sparse_key_names,
+      const std::vector<std::string>& sparse_grad_names, const int emb_dim,
+      std::vector<std::vector<float>>* push_values,
+      std::vector<::std::future<int32_t>>* push_sparse_status);
+
+  // Push sparse variables to server in Async mode
+  // Param<In>: scope, table_id, fea_keys, sparse_grad_names
+  // Param<Out>: push_values, push_sparse_status
+  /*
+  void PushSparseVarsAsync(
+          const Scope& scope,
+          const uint64_t table_id,
+          const std::vector<uint64_t>& fea_keys,
+          const std::vector<std::string>& sparse_grad_names,
+          std::vector<std::vector<float>>* push_values,
+          std::vector<::std::future<int32_t>>* push_sparse_status);
+  */
+
+  void InitServer(const std::string& dist_desc, int index);
+  void InitWorker(const std::string& dist_desc,
+                  const std::vector<uint64_t>& host_sign_list, int node_num,
+                  int index);
+  void StopServer();
+  uint64_t RunServer();
+  void GatherServers(const std::vector<uint64_t>& host_sign_list, int node_num);
+
+  static std::shared_ptr<FleetWrapper> s_instance_;
+  static std::shared_ptr<FleetWrapper> GetInstance() {
```

Comment:
Maybe we should put `static std::shared_ptr<FleetWrapper> s_instance_` a static local variable inside `GetInstance()`, to ensure a singleton

Reply:
done.